### DISCUSSION TITLE
I want to store nested comments, like on reddit. How can I retrieve all the nested comments?

### DISCUSSION THREAD

**Discussion by lumenwrites** [prisma/prisma#10406] [Q&A] [answered]
Hi! I want to model nested comments, like on reddit. I'm using a one-to-many self relation like this:
```
model Comment {
  [...]
  parentId String?
  parent Comment? @relation("ParentChildren", fields: [parentId], references: [id])
  children Comment[] @relation("ParentChildren")
}
```

So each child is connected to its parent by `parentId`. The comments can be infinitely nested, parents can have grand-children, grand-grand-children, and so on.

My question is - is it possible to retrieve all of the parent comment's descendants in one query? How can I do that?

My goal is to get a json object that looks kinda like this:

```
  {
    comment: 'Lorem ipsum...',
    children: [
      {
        comment: 'Lorem ipsum...',
        children: [
          {
            comment: 'Lorem ipsum...',
            children: []
          },
          {
            comment: 'Lorem ipsum...',
            children: []
          },
        ],
      },
    ],
  },
```

**Comment by TasinIshmam** [comment, marked as answer]
Hey @lumenwrites 

Reposting my response from the [stackoverflow question](https://stackoverflow.com/questions/70095411/i-want-to-use-prisma-to-store-nested-comments-like-on-reddit-how-can-i-retrie), in case it helps someone else. 

Unfortunately, this isn't supported at the moment. The only possible workaround would be to implement it in SQL using `rawQuery`. There's an [issue](https://github.com/prisma/prisma/issues/3725)  about it in the Prisma repo.

**Reply by skorotkiewicz** [reply]
[Solutions](https://github.com/prisma/prisma/discussions/10406#discussioncomment-2577645)

**Comment by skorotkiewicz** [comment]
```javascript
model Comment {
  id        Int      @id @default(autoincrement())
  comment   String

  Children  Comment[] @relation("Comment_Children")
  parent    Comment?  @relation("Comment_Children", fields: [parent_id], references: [id])
  parent_id Int?
}
```

```javascript
let comment = await prisma.comment.findMany({
  where: {
    parent_id: null,
  },
  include: {
    children: {
      include: {
        children: true,
      },
    },
  },
});
```

```javascript
let comment = await prisma.comment.create({
  data: {
    comment: req.body.comment,
    parent_id: req.body.parent_id, // or `null` for parent comment
  },
  include: {
    Children: {
      include: {
        Children: true,
      },
    },
  },
});
```

**Reply by Umbrien** [reply]
The suggested use of `findMany` will only search for items at a depth of one level

**Reply by Ashfak-Hossain** [reply]
will it work for infinity nested nodes ? or just 2 level ?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
